rust/crd/src/resource.rs

```diff
@@ -77,6 +83,43 @@ impl RoleResource {
             Self::Historical(r) => r.clone().memory,
         }
     }
+
+    /// Update the given configuration file with resource properties.
+    /// Currently it only adds the segment cache location property for historicals to runtime.properties.
+    pub fn update_druid_config_file(
+        &self,
+        file: &str,
+        config: &mut BTreeMap<String, Option<String>>,
+    ) {
+        if let Self::Historical(r) = self {
+            if RUNTIME_PROPS == file {
+                let free_percentage = r.storage.segment_cache.free_percentage.unwrap_or(5u16);
```

Comment:
@adwk67 added on slack to the comment above:
> What I mean is: I thought one of the main points of the fragments was to be able to merge things, including defaults. But I also read somewhere (can't find it now) that operator-rs 0.26 doesn't consider all enums. Will we - with a later operator-rs version - be able to read from a fragment, knowing that the defaults have been applied where relevant?

Defaults are merged here already and that `unwrap_or()` is never (semantically) needed, but Rust doesn't know that. So as long as `free_percentage` is an `Optional`, we will always have to cater for the case where it could theoretically be `None`.

Reply:
Thanks for the clarification!